File: Observer/Dart/MaxObserver.dart
import 'IObserver.dart';
import 'dart:math';

class MaxObserver implements IObserver {
  int _maxCPU = 0, _maxRAM = 0;

  @override
  void update(int CPU, int RAM) {
    _maxCPU = max(_maxCPU, CPU);
    _maxRAM = max(_maxRAM, RAM);

    print("""
Current Max CPU: $_maxCPU
Current Max RAM: $_maxRAM
""");
  }
}

File: Observer/Javascript/MaxObserver.js
import IObserver from "./IObserver.js";

class MaxObserver extends IObserver {
  constructor() {
    super();
    this.maxCPU = 0;
    this.maxRAM = 0;
  }

  update(cpu, ram) {
    this.maxCPU = Math.max(this.maxCPU, cpu);
    this.maxRAM = Math.max(this.maxRAM, ram);

    console.log(`Current Max CPU: ${this.maxCPU}\nCurrent Max RAM: ${this.maxRAM}`);
  }
}

export default MaxObserver;